core/kazoo_data/src/kzs_util.erl

```diff
@@ -111,51 +112,107 @@ bind_db_classify(Database, M, F) ->
-db_priority(?MATCH_ACCOUNT_UNENCODED(_AccountId)) -> 21;
-db_priority(?MATCH_ACCOUNT_encoded(_AccountId)) -> 21;
-db_priority(?MATCH_ACCOUNT_ENCODED(_AccountId)) -> 21;
-db_priority(?MATCH_MODB_SUFFIX_UNENCODED(_A,_B,_Rest,_Year,_Month)) -> 22;
-db_priority(?MATCH_MODB_SUFFIX_ENCODED(_A,_B,_Rest,_Year,_Month)) -> 22;
-db_priority(?MATCH_MODB_SUFFIX_encoded(_A,_B,_Rest,_Year,_Month)) -> 22;
-db_priority(?MATCH_MODB_SUFFIX_RAW(_Account,_Year,_Month)) -> 22;
-db_priority(?MATCH_RESOURCE_SELECTORS_UNENCODED(_AccountId)) -> 23;
-db_priority(?MATCH_RESOURCE_SELECTORS_encoded(_AccountId)) -> 23;
-db_priority(?MATCH_RESOURCE_SELECTORS_ENCODED(_AccountId)) -> 23;
-db_priority(?MATCH_RESOURCE_SELECTORS_RAW(_AccountId)) -> 23;
-db_priority(?MATCH_PROVISIONER_ENCODED(_AccountId)) -> 24;
-db_priority(?MATCH_PROVISIONER_encoded(_AccountId)) -> 24;
-db_priority(_Database) -> 24.
+db_priority(Db) ->
+    db_priority(Db, modb_priority()).
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec db_priority(kz_term:text(), map()) -> non_neg_integer().
+db_priority(Db, MODBPriority) when not is_binary(Db) ->
+    db_priority(kz_term:to_binary(Db), MODBPriority);
+db_priority(?KZ_CONFIG_DB, _) -> 0;
+db_priority(?KZ_DATA_DB, _) -> 1;
+db_priority(?KZ_OFFNET_DB, _) -> 2;
+db_priority(?KZ_ACCOUNTS_DB, _) -> 3;
+db_priority(?KZ_SIP_DB, _) -> 4;
+db_priority(?KZ_AUTH_DB, _) -> 5;
+db_priority(?KZ_WEBHOOKS_DB, _) -> 6;
+db_priority(?KZ_RATES_DB, _) -> 7;
+db_priority(?KZ_ACDC_DB, _) -> 8;
+db_priority(?KZ_FAXES_DB, _) -> 9;
+db_priority(?KZ_SCHEMA_DB, _) -> 10;
+db_priority(?KZ_SERVICES_DB, _) -> 11;
+db_priority(?KZ_PORT_REQUESTS_DB, _) -> 12;
+db_priority(?KZ_TASKS_DB, _) -> 13;
+db_priority(?KZ_PENDING_NOTIFY_DB, _) -> 13;
+db_priority(?KZ_DEDICATED_IP_DB, _) -> 14;
+db_priority(?KZ_ALERTS_DB, _) -> 15;
+db_priority(?KZ_MEDIA_DB, _) -> 16;
+db_priority(?KZ_OAUTH_DB, _) -> 17;
+db_priority(?KZ_TOKEN_DB, _) -> 18;
+db_priority(<<?KNM_DB_PREFIX, _/binary>>, _) -> 19;
+db_priority(<<?KNM_DB_PREFIX_ENCODED, _/binary>>, _) -> 19;
+db_priority(<<?KNM_DB_PREFIX_encoded, _/binary>>, _) -> 19;
+db_priority(?MATCH_ACCOUNT_UNENCODED(_AccountId), _) -> 21;
+db_priority(?MATCH_ACCOUNT_encoded(_AccountId), _) -> 21;
+db_priority(?MATCH_ACCOUNT_ENCODED(_AccountId), _) -> 21;
+db_priority(?MATCH_MODB_SUFFIX_UNENCODED(_A,_B,_Rest,_Year,_Month) = Db, MODBPriority) ->
+    db_priority_modb(kz_util:format_account_modb(Db, 'raw'), MODBPriority);
+db_priority(?MATCH_MODB_SUFFIX_ENCODED(_A,_B,_Rest,_Year,_Month) = Db, MODBPriority) ->
+    db_priority_modb(kz_util:format_account_modb(Db, 'raw'), MODBPriority);
+db_priority(?MATCH_MODB_SUFFIX_encoded(_A,_B,_Rest,_Year,_Month) = Db, MODBPriority) ->
+    db_priority_modb(kz_util:format_account_modb(Db, 'raw'), MODBPriority);
+db_priority(?MATCH_MODB_SUFFIX_RAW(_Account,_Year,_Month), _) -> 22;
```

Comment:
seems to be missing here the call to ` db_priority_modb(kz_util:format_account_modb(Db, 'raw'), MODBPriority);`